code/script.js

```diff
@@ -215,111 +220,197 @@ const generateBoard = () => {
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+    }
+  }
+  generateBoard(charactersInPlay)
+  // I encounter some problems of the above block of code, I used chatGPT to check what was wrong. It provided an improvement of my code
+  // Although I realized the problem was the value in the drop down list (hats) didn't match the attribube in the object (hat)
+  // const filterCondition = keep
+  //   ? (person) => person[category].includes(value)
+  //   : (person) => !person[category].includes(value);
+
+  // if (category === 'accessories') {
+  //   if (keep) {
+  //     alert(`Yes, the person wears ${value}! Keep all people that wear ${value}`);
+  //   } else {
+  //     alert(`No, the person doesn't wear ${value}! Remove all people that wear ${value}`);
+  //   }
+
+  //   charactersInPlay = charactersInPlay.filter(filterCondition);
+  // }
 
-  // Invoke a function to redraw the board with the remaining people.
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// const redraw = 
+
+// Determine what is the category
+// filter by category to keep or remove based on the keep variable.
+/* 
+  for hair and eyes :
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+    or
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+ 
+  for accessories and other
+    charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+    or
+    charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+*/
+
+// Invoke a function to redraw the board with the remaining people.e player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  // personToConfirm = person.name
+  if (confirm(`Are you sure you want to guess on ${personToConfirm}?`) == true) {
+    checkMyGuess(personToConfirm)
+  }
 }
+// store the interaction from the player in a variable.
+// remember the confirm() ?
+// If the player wants to guess, invoke the checkMyGuess function.
+
 
 // If you confirm, this function is invoked
+
+
+// 1. Check if the personToCheck is the same as the secret person's name
+// 2. Set a Message to show in the win or lose section accordingly
+// 3. Show the win or lose section
+// 4. Hide the game board
+
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  if (personToCheck === secret.name) {
+    winOrLose.style.display = "flex";
+    winOrLoseText.innerHTML += `
```

Comment:
I suggest writing only "=" here as you would like to replace the text with a new one depending on the result, not adding to it. I got once both win and lose messages when I guessed right.

Reply:
Yes you are so right!!! Thanks !